src/baobab.d.ts

```diff
@@ -0,0 +1,205 @@
+import Emitter from "emmett";
+
+interface PlainObject<T = any> {
+  [key: string]: T;
+}
+
+type Predicate = (data: any) => boolean;
+type Constraints = object;
+type PathKey = string | number;
+type PathElement = PathKey | Predicate | Constraints;
+export type Path = PathElement[] | PathKey;
+
+type Splicer = [number | object | ((...args: any[]) => any), ...any[]];
+
+/**
+ * This class is empty purposely. Baobab must be able to identify in an initial
+ * state when it has to deal with Monkeys instanciation, and uses this dummy
+ * class in that purpose.
+ */
+export class MonkeyDefinition {
+  // Empty class intended
+}
+
+export class Monkey {
+  // TODO
```

Comment:
I should probably add types for `Monkey`, since it actually leaks into the public API through `Baobab#getMonkey`.